### PR TITLE
Crop to overlap and other improvements

### LINKED ISSUE #76: Pass all tests
The module currently fails some tests when `ITKMontage_EnableFailingTests` is turned on. Improvements need to be made to pass this.

### LINKED ISSUE #95: Crop image to overlapping portion for registration
We should enlarge the cropping area, e.g. + 1% of image size + 5% of overlap + 16 pixels.

This will improve the brightness of the true peak when image content is repetitive or monotonous. Additionally, it should speed up the computation as less pixels need to be processed in common cases.

### LINKED ISSUE #114: Change default FFT type from double to float
With char and float inputs, it will probably not make a difference in precision. It should make a reduction in processing time, at the very least due to less memory traffic.

### PR DESCRIPTION
**Pull request by dzenanz**
Closes #114. Closes #95. Closes #76.

### CODE REVIEW

File: examples/PhaseCorrelationImageRegistration.cxx [hunk 1]
```diff
@@ -72,12 +72,12 @@ int main( int argc, char *argv[] )
     }
 
   // init the registration method
-  using RegistrationType = itk::PhaseCorrelationImageRegistrationMethod< ImageType, ImageType >;
+  using RegistrationType = itk::PhaseCorrelationImageRegistrationMethod< ImageType, ImageType, float >;
   RegistrationType::Pointer pcmRegistration = RegistrationType::New();
   pcmRegistration->SetFixedImage( fixedReader->GetOutput() );
   pcmRegistration->SetMovingImage( movingReader->GetOutput() );
 
-  using OperatorType = itk::PhaseCorrelationOperator< itk::NumericTraits< PixelType >::RealType, Dimension >;
```
Comment (thewtex) [inline examples/PhaseCorrelationImageRegistration.cxx]: Why force `float`?

File: include/itkMaxPhaseCorrelationOptimizer.hxx [hunk 2]
```diff
@@ -163,64 +163,57 @@ MaxPhaseCorrelationOptimizer< TRegistrationMethod >
...
+            }
           }
-        }
-    },
-    nullptr );
+      },
+      nullptr );
 
-  WriteDebug( iAdjusted.GetPointer(), "iAdjustedZS.nrrd" );
+    WriteDebug( iAdjusted.GetPointer(), "iAdjustedZS.nrrd" );
```
Comment (thewtex) [inline include/itkMaxPhaseCorrelationOptimizer.hxx]: Debug code

File: include/itkPhaseCorrelationImageRegistrationMethod.h [hunk 3]
```diff
@@ -94,13 +95,18 @@ namespace itk
...
+ *
  * \author Jakub Bican, jakub.bican@matfyz.cz, Department of Image Processing,
  *         Institute of Information Theory and Automation,
  *         Academy of Sciences of the Czech Republic.
  *
  * \ingroup Montage
  */
-template< typename TFixedImage, typename TMovingImage >
+template< typename TFixedImage, typename TMovingImage,
+  typename TInternalPixelType = typename std::conditional<
```
Comment (thewtex) [inline include/itkPhaseCorrelationImageRegistrationMethod.h]: Same type logic should be applied in the example?

File: examples/PhaseCorrelationImageRegistration.cxx (hunk 1 above)
Comment (dzenanz) [reply examples/PhaseCorrelationImageRegistration.cxx]: Results are pretty much the same, but float is some 10% or so faster.

File: include/itkMaxPhaseCorrelationOptimizer.hxx (hunk 2 above)
Comment (dzenanz) [reply include/itkMaxPhaseCorrelationOptimizer.hxx]: WriteDebug only writes in debug mode. It does nothing in release mode. See [definition](https://github.com/InsightSoftwareConsortium/ITKMontage/blob/1962de8ab6a00eb698a5a3716a9c6d5d04eb167d/include/itkPhaseCorrelationOptimizer.hxx#L23-L51). We should clean up debugging code as one of the final things. I added an [issue](https://github.com/InsightSoftwareConsortium/ITKMontage/issues/117) to track that.

File: include/itkPhaseCorrelationImageRegistrationMethod.h (hunk 3 above)
Comment (dzenanz) [reply include/itkPhaseCorrelationImageRegistrationMethod.h]: In the library code, we don't know which type will be instantiated. In the examples we have concrete types.

**Comment by dzenanz** [comment]
```text
Object
  SHA512=b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb
  not found at:

    https://insightsoftwareconsortium.github.io/ITKTestingData/SHA512/b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb ("HTTP response code said error")
    https://data.kitware.com:443/api/v1/file/hashsum/SHA512/b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb/download ("Server returned nothing (no headers, no data)")
    https://itk.org/files/ExternalData/SHA512/b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb ("HTTP response code said error")
    https://slicer.kitware.com/midas3/api/rest?method=midas.bitstream.download&checksum=b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb&algorithm=SHA512 (wrong hash SHA512=639526ba1271b833beeb026311a3d40d2b50b8480ce8e938e87ea0dfc9c84e76867be09043f9b15fc5b21d003580c6169f1ba59b2202686b80258d8993ab7a09)
```
I can download it from [data.kitware.com](https://data.kitware.com:443/api/v1/file/hashsum/SHA512/b7f598eb3bc8b2b48908d6d48dcb274cbc3a0ff17331178db40919218b68f3dcc3c39f67bc1a8c9c001c223f72f25a93452bfc26f5bd4af4af7f8aec05a369bb/download), it is located [here](https://data.kitware.com/#item/5c756c818d777f072b6885b2). I don't know why CI machine fails there.

**Comment by dzenanz** [comment]
The performance generally improves slightly. A big increase in duration of synthetic tests murkies the water somewhat.
[MontageBefore.txt](https://github.com/InsightSoftwareConsortium/ITKMontage/files/3346468/MontageBefore.txt)
[MontageAfter.txt](https://github.com/InsightSoftwareConsortium/ITKMontage/files/3346469/MontageAfter.txt)

**Pull request merged**